### DISCUSSION TITLE
Lesson 10 9:23:43 Why is my transaction reverting?

### DISCUSSION THREAD

**Discussion by delphee** [smartcontractkit/full-blockchain-solidity-course-py#476] [Q&A] [answered]
`PS C:\Users\The Elphees\aave_brownie_py> brownie run scripts/aave_borrow.py
INFO: Could not find files for the given pattern(s).
Brownie v1.17.0 - Python development framework for Ethereum

AaveBrowniePyProject is the active project.

Launching 'ganache-cli.cmd --accounts 10 --fork https://eth-mainnet.alchemyapi.io/v2/yuSD4lrhnilT3jifb-RJk8rLTLcFwCHv --mnemonic brownie --port 8545 --hardfork istanbul'...

Running '\Users\The Elphees\aave_brownie_py\scripts\aave_borrow.py::main'...
mainnet-fork
mainnet-fork
Transaction sent: 0x7657b5954178ce7f6c1ef9a2cc479ee517a1241fb8c012fe67865098c10e0da0
  Gas price: 0.0 gwei   Gas limit: 6721975   Nonce: 2
  Transaction confirmed   Block: 13730510   Gas used: 43738 (0.65%)

  Transaction confirmed   Block: 13730510   Gas used: 43738 (0.65%)

Received 0.1 WETH
lending pool: 0xB53C1a33016B2DC2fF3653530bfF1848a515c8c5
Approving ERC20 token...
Transaction sent: 0xcf2e4e1a6c6a7bfbc6a472f245bad131486cd82fe4c2c8013505330d978673ef
  Gas price: 0.0 gwei   Gas limit: 6721975   Nonce: 3
  Transaction confirmed   Block: 13730511   Gas used: 43952 (0.65%)

  Transaction confirmed   Block: 13730511   Gas used: 43952 (0.65%)

Approved!
depositing
Transaction sent: 0xf9c456798dd66acee1d4fe77ff4231dc4c27b794ca7c7e897da8378d934e8f2c
  Gas price: 0.0 gwei   Gas limit: 6721975   Nonce: 4
  Transaction confirmed (reverted)   Block: 13730512   Gas used: 22397 (0.33%)

  File "C:\Users\The, line line, in in
    return_value, frame = run(
  File "C:\Users\The, line line, in in
    return_value = f_locals[method_name](*args, **kwargs)
  File "\Users\The, line line, in in
    tx = lending_pool.deposit(
  File "C:\Users\The, line line, in in
    return self.transact(*args)
  File "C:\Users\The, line line, in in
    return tx["from"].transfer(
  File "C:\Users\The, line line, in in
    receipt._raise_if_reverted(exc)
  File "C:\Users\The, line line, in in
    raise exc._with_attr(
VirtualMachineError: revert
Terminating local RPC client...`

![image](https://user-images.githubusercontent.com/38186979/144525530-6a667b6b-1d83-43d9-a485-73b2dd3943d5.png)
![image](https://user-images.githubusercontent.com/38186979/144525653-b258754f-4c35-4fdf-80c2-36172fbdfe9e.png)

**Comment by cromewar** [comment, marked as answer]
hello @delphee please could you share your aave_borrow.py in order to check it, also make sure the `get_lending_pool` function is like this:

```Python

def get_lending_pool():
    lending_pool_addresses_provider = interface.ILendingPoolAddressesProvider(
        config["networks"][network.show_active()]["lending_pool_addresses_provider"]
    )
    lending_pool_address = lending_pool_addresses_provider.getLendingPool()
    lending_pool = interface.ILendingPool(lending_pool_address)
    return lending_pool
```

**Comment by delphee** [comment]
cromewar,  it was a typo in my get_lending_pool().
Thank you, again.
I really do try... before asking for help.
Thank you for having patience with a beginner.

**Reply by cromewar** [reply]
Is my pleasure, I'm glad you solved the problem. And feel free to make any other question here I'll do my best to help you.

Best regards, Cromewar.

**Reply by WeTheArtist** [reply]
I had the same error and I checked my code as well, there were no spelling mistakes, but I anyways copied your function and pasted it. And it worked!
I was then doubtful if it was a spelling error so I pasted the function that I had written and ran it again and it worked !
I also compared the function that you mentioned and the one that I had written from the video and it was identical.
Really strange to see that. Does this happen from time to time?

**Reply by cromewar** [reply]
Not quite sure as it must work every time the code is correct, may be this could be a identation problem and pasting the functions fixes it.
